### PR TITLE
dupeguru: 4.0.4 -> 4.1.1

### LINKED ISSUE #120205: dupeguru: 4.0.4 -> 4.1.1
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: nixpkgs.dupeguru

current version: 4.0.4
desired version: 4.1.1

###### Notify maintainers

maintainers: @novoxudonoser

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
#Closes https://github.com/NixOS/nixpkgs/issues/120205

###### Things done
I've tried to make the checks work, I've opened an issue here https://github.com/arsenetar/dupeguru/issues/875.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dschrempf** [comment]
I suggest replacing the former maintainer `novoxudonoser`. I can't find her/him on GitHub anymore.

**Comment by SCOTT-HAMILTON** [comment]
Hello, @novoxd could you please confirm dschrempf's above-hypothesis ?

**Comment by austinbutler** [comment]
Also fixes https://github.com/NixOS/nixpkgs/issues/120203.

@SCOTT-HAMILTON are you still planning on finishing this?

**Comment by SCOTT-HAMILTON** [comment]
Thanks for the reminder !

File: pkgs/applications/misc/dupeguru/default.nix [hunk 1]
```diff
@@ -58,14 +61,29 @@ python3Packages.buildPythonApplication rec {
...
+    black
+    distro
+    flake8
+    pytest
+    send2trash
+    tox
+  ];
+
+  checkPhase = ''
+    tox
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:79]: Can we execute pytest through pytestCheckHook?

File: pkgs/applications/misc/dupeguru/default.nix [hunk 2]
```diff
@@ -58,14 +61,29 @@ python3Packages.buildPythonApplication rec {
...
   # so wrapPythonPrograms hook does not handle it automatically.
   postFixup = ''
-    wrapPythonProgramsIn "$out/share/dupeguru" "$out $pythonPath"
+    wrapPythonProgramsIn "$out/share/dupeguru" "$out $propagatedBuildInputs"
+  '';
+
+  doCheck = true;
+
+  checkInputs = [
+    black
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:70]:
```suggestion
```

File: pkgs/applications/misc/dupeguru/default.nix [hunk 3]
```diff
@@ -58,14 +61,29 @@ python3Packages.buildPythonApplication rec {
...
+    distro
+    flake8
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:72]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/dupeguru/default.nix [hunk 4]
```diff
@@ -1,54 +1,57 @@
...
 
   format = "other";
 
   src = fetchFromGitHub {
     owner = "arsenetar";
     repo = "dupeguru";
     rev = version;
-    sha256 = "0ma4f1c6vmpz8gi4sdy43x1ik7wh42wayvk1iq520d3i714kfcpy";
+    sha256 = "1a34kkw2xyfdirv0nw5w6v7s2db4cfn17ksnnblkl7ahndr6hlnj";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:30]: There are no msubodules in the repo.

**Comment by SuperSandro2000** [comment]
> Hi! Sorry for bumbing this, #120203 and #120205 were marked stale.

Please open a new PR if the author is not responding and you want to see the change happening.

File: pkgs/applications/misc/dupeguru/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/dupeguru/default.nix]: Fixed.

File: pkgs/applications/misc/dupeguru/default.nix (hunk 4 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/dupeguru/default.nix]: Fixed.

**Comment by SCOTT-HAMILTON** [comment]
> Hi! Sorry for bumbing this, #120203 and #120205 were marked stale.

Sorry for the delay, I've been procrastinating on this for too long.

File: pkgs/applications/misc/dupeguru/default.nix [hunk 5]
```diff
@@ -58,14 +59,28 @@ python3Packages.buildPythonApplication rec {
...
-    wrapPythonProgramsIn "$out/share/dupeguru" "$out $pythonPath"
+    wrapPythonProgramsIn "$out/share/dupeguru" "$out $propagatedBuildInputs"
+  '';
+
+  doCheck = true;
+
+  checkInputs = [
+    pytestCheckHook
+    distro
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:70]:
```suggestion
```
Already included in pytestCheckHook.

File: pkgs/applications/misc/dupeguru/default.nix [hunk 6]
```diff
@@ -58,14 +59,28 @@ python3Packages.buildPythonApplication rec {
...
+    send2trash
+    tox
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dupeguru/default.nix:72]: Do we really need tox? Did you try without?

Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/dupeguru/default.nix]: Fixed, we actually didn't.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 120322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dupeguru</li>
  </ul>
</details>

**Comment by austinbutler** [comment]
Builds and works for me.

Looks like the `preFixup` can be deleted, the linked bug was fixed in https://github.com/NixOS/nixpkgs/pull/76283.

File: pkgs/applications/misc/dupeguru/default.nix [hunk 7]
```diff
@@ -58,14 +56,26 @@ python3Packages.buildPythonApplication rec {
...
+  doCheck = true;
+
+  checkInputs = [
+    pytestCheckHook
+    distro
+    send2trash
+  ];
+
+  preCheck = ''
+    export HOME=$(pwd)
```
Comment (austinbutler) [inline pkgs/applications/misc/dupeguru/default.nix:71]: Maybe it's obvious to others, but I'd appreciate a comment explaining why this is needed.

Comment (austinbutler) [reply pkgs/applications/misc/dupeguru/default.nix:71]: Looks like the Python tests try to write to the `$HOME` folder. The manual seems to suggest `export HOME=$(mktemp -d)`, but it seems this also works.

https://github.com/NixOS/nixpkgs/blob/a0bf6b28dff054b4d95f28ff7e2d7b99b60aa027/doc/languages-frameworks/python.section.md?plain=1#L1258

**Review by austinbutler — APPROVED** [review]

File: pkgs/applications/misc/dupeguru/default.nix (hunk 7 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/misc/dupeguru/default.nix]: Fixed.

**Comment by SCOTT-HAMILTON** [comment]
> Builds and works for me.
> 
> Looks like the `preFixup` can be deleted, the linked bug was fixed in #76283.
> 
> EDIT: Removing the `preFixup` doesn't work, the resulting bin file is missing the various `export QT_` variables, so it fails to start. Only suggestion then would be removing the link to that conversation as it seems the cause isn't that particular bug.

Yep, I kept it and updated the comment.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 120322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dupeguru</li>
  </ul>
</details>

**Review by austinbutler — APPROVED** [review]

**Comment by doronbehar** [comment]
Closed due to #160107.

**Pull request closed without merging**